pkg/metadata/inventories/payload.go

```diff
@@ -46,35 +38,5 @@ func (p *Payload) MarshalJSON() ([]byte, error) {
 // In this case, the payload can only be split at the top level, so `times` is ignored
 // and each top-level component is returned as a distinct payload.
 func (p *Payload) SplitPayload(times int) ([]marshaler.AbstractMarshaler, error) {
-	newPayloads := []marshaler.AbstractMarshaler{}
-	fieldName := ""
-
-	// Each field can be sent individually but we can't split any more than this as the backend expects each payload
-	// to be received complete.
-
-	if p.CheckMetadata != nil {
-		fieldName = "check_metadata"
-		newPayloads = append(newPayloads,
-			&Payload{
-				Hostname:      p.Hostname,
-				Timestamp:     p.Timestamp,
-				CheckMetadata: p.CheckMetadata,
-			})
-	}
-	if p.HostMetadata != nil {
-		fieldName = "host_metadata"
-		newPayloads = append(newPayloads,
-			&Payload{
-				Hostname:     p.Hostname,
-				Timestamp:    p.Timestamp,
-				HostMetadata: p.HostMetadata,
-			})
-	}
-
-	// if only one field is set we can't split any more
-	if len(newPayloads) <= 1 {
-		return nil, fmt.Errorf("could not split inventories payload any more, %s metadata is too big for intake", fieldName)
-	}
-
-	return newPayloads, nil
+	return nil, fmt.Errorf("could not split inventories payload any more, payload is too big for intake")
```

Comment:
We use to send 3 blob of data in one payload (`host`, `agent` and `checks`). Now that we only send one it can no longer be split (the endpoint doesn't support receiving metadata for a type in chunks).